experiments/mountain_car_plots.py

```diff
@@ -0,0 +1,269 @@
+  """
+  assert 0 <= act <= 2
+
+  obs_vec, acts_vec = [], []
+  for p in np.arange(MC_POS_MIN, MC_POS_MAX + 1e-5, step=pos_step_size):
+    for v in np.arange(MC_VEL_MIN, MC_VEL_MAX + 1e-5, step=velocity_step_size):
+      obs_vec.append([p, v])
+      acts_vec.append(act)
+
+  obs_vec = np.array(obs_vec)
+  acts_vec = np.array(acts_vec)
+  next_obs_vec = make_next_obs(obs_vec, acts_vec)
+
+  R = reward_fn(obs_vec, acts_vec, next_obs_vec, None)
+
+  df = pd.DataFrame()
+  df["position"] = np.round(obs_vec[:, 0], 5)
+  df["velocity"] = np.round(obs_vec[:, 1], 5)
+  df["reward"] = R
+  df = df.pivot("velocity", "position", "reward")
+
+  def convert_to_rc(pos, vel):
+    x = (np.array(pos) - MC_POS_MIN) / pos_step_size + 0.5
+    y = (np.array(vel) - MC_VEL_MIN) / velocity_step_size + 0.5
+    return x, y
+
+  def convert_to_rc_filtered(trajs):
+    trans = util.rollout.flatten_trajectories(trajs)
+    obs = trans.obs
+    if filter_trans_by_act:
+      obs = obs[trans.acts == act]
+    pos, vel = obs[:, 0], obs[:, 1]
+    return convert_to_rc(pos, vel)
+
+  plt.figure()
+  if heatmap:
+    sns.heatmap(df)
+
+  if mark_goal:
+    goal_r, _ = convert_to_rc(MC_GOAL_POS, 0)
+    plt.axvline(x=goal_r, linestyle='--',
+                label=f"goal state (pos={MC_GOAL_POS})")
+  if exp_trajs is not None:
+    X, Y = convert_to_rc_filtered(exp_trajs)
+    plt.scatter(X, Y, marker="o", label="expert samples", alpha=0.2)
+  if gen_trajs is not None:
+    X, Y = convert_to_rc_filtered(gen_trajs)
+    plt.scatter(X, Y, marker="o", c="yellow", label="policy samples",
+                alpha=0.2)
+
+  act_names = ["left", "neutral", "right"]
+  if title is None:
+    title = f"Action {act_names[act]}"
+  plt.title(title)
+  if legend_on:
+    plt.legend(loc="center left", bbox_to_anchor=(0, 1.3))
+
+
+def batch_reward_heatmaps(
+    checkpoints_dir: Union[str, Path],
+    output_dir: Union[str, Path] = Path("/tmp/default"),
+    exp_trajs: Optional[List[util.rollout.Trajectory]] = None,
+) -> None:
+  """Save mountain car reward heatmaps for every action and every checkpoint.
+
+  Plots with rollout transitions scatterplotted on top are saved in
+  "{output_dir}/act_{i}".
+  Plots without rollout transitions are saved in
+  "{output_dir}/no_rollout_act_{i}".
+
+  Args:
+      checkpoints_dir: Path to `checkpoint` directory from AIRL or GAIL output
+          directory. Should contain "gen_policy" and "discrim" directories.
+      output_dir: Heatmap output directory.
+      exp_trajs: Expert trajectories for scatterplotting.
+          Generator trajectories
+          are dynamically generated from generator checkpoints.
+  """
+  checkpoints_dir = Path(checkpoints_dir)
+  for checkpoint_dir in sorted(checkpoints_dir.iterdir()):
+    vec_normalize_path = checkpoint_dir / "gen_policy" / "vec_normalize.pkl"
+    discrim_path = checkpoint_dir / "discrim"
+    policy_path = checkpoint_dir / "gen_policy"
+
+    with open(vec_normalize_path, "rb") as f:
```

Comment:
I suggest moving this after `gen_trajs`, and adding a comment to the effect that `make_policy_rollout` automatically loads normalization statistics. (For a while I was puzzled by why we weren't using `vec_normalize` anywhere in the rollouts.)